db/migrate/20220105085729_split_customers_name.rb

```diff
@@ -0,0 +1,22 @@
+class SplitCustomersName < ActiveRecord::Migration[6.1]
+  class Customer < ActiveRecord::Base
+  end
+
+  def change
+    add_column :customers, :first_name, :string
+    add_column :customers, :last_name, :string
+    rename_column :customers, :name, :backup_name
+    reversible do |dir|
+      dir.up { migrate_customer_name_data }
+    end
+  end
+
+  def migrate_customer_name_data
+    Customer.where("backup_name LIKE '% %'").find_each do |customer|
+      first_name, last_name = customer.backup_name.split(' ', 2)
+      customer.first_name = first_name
+      customer.last_name = last_name
+      customer.save
```

Comment:
I think it'd be good to break down the available data into sets and handle each set as optimally as we can, splitting the names in multiple passes. One thing we can do: a large percentage of customers records have a `bill_address.first_name` and `bill_address.last_name` that match the `customer.name` but are already correctly split into two columns. We should favour that option where it's available, as it'll give the highest fidelity. This is important where we have names that currently contain three or more parts like `Shia Le Bouf` or `Mary Jane Watson` where it's not possible to determine whether or we need to split on the first or second space.

Another thing we could do just before dealing with the data is normalize it a bit. There are a lot of entries in the `customer.name` column which have superfluous spaces in them like "Alice   &nbsp; &nbsp;  Jones". If we strip them out in a first pass before handling them, it'll be easier to deal with.

Reply:
Ok, got it, I will use the `bill_address.first_name` and `bill_address.last_name` in priority, and if not present apply the current process with `customer.name` value. Any new value will be stripped before being saved.